### PR TITLE
ready_for_review datadog_event module:   Datadog API http status code 202 is ok.   Ref: http://docs.datadoghq.com/api/

### PR DESCRIPTION
**Pull request by n0ts**
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

monitor/datadog_event
##### ANSIBLE VERSION

```
2.1.0.0
```
##### SUMMARY

datadog event module status code 202 is ok.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @n0ts. To the current maintainers, @arturaz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @n0ts for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
@gregdek  Thanks, I rebase it.
ready_for_review

**Comment by gregdek** [comment]
Thanks @n0ts. To the current maintainers, @arturaz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by arturaz** [comment]
Hey there. We're not using datadog and I have no interest in maintaining this module. How can I unsubscribe from this module?

**Comment by n0ts** [comment]
@gregdek @arturaz ok, I'll continue to maintainers.

**Comment by gregdek** [comment]
@n0ts thanks for volunteering to help with this module! I have made you the new maintainer.

**Comment by gregdek** [comment]
Thanks again to @n0ts. This module is going into community review.

We notice that you are one of the maintainers (@n0ts) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
shipit it ok

**Comment by gregdek** [comment]
Thanks @n0ts. Since you are one of the maintainers (@n0ts) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
@gregdek Thanks, "shipit"

File: monitoring/datadog_event.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def post_event(module):
     headers = {"Content-Type": "application/json"}
 
     (response, info) = fetch_url(module, uri, data=json_body, headers=headers)
-    if info['status'] == 202:
+    if (info['status'] == 200) in (info['status'] == 202):
```
Comment (resmo) [inline monitoring/datadog_event.py]: I don't see the sense in this condition, shouldn't it be more like

``` python
if info['status'] in [200, 202]:
```

**Comment by resmo** [comment]
needs_info

**Comment by gregdek** [comment]
@n0ts, a community member requested some information about this PR; for that reason, we put this PR into the 'needs_info' state.

When you have answered the questions, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@n0ts A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@n0ts Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
@gregdek @resmo Thanks for your comment. I use official datadog create event api. Could you review it?

**Comment by gregdek** [comment]
Thanks @n0ts for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
@greenbrian Thanks a lot! I rebase it.

`ready_for_review`

**Comment by n0ts** [comment]
@gregdek @resmo 

ready_for_review

**Comment by n0ts** [comment]
@gregdek 

`ready_for_review`

**Comment by gregdek** [comment]
Thanks @n0ts. To the current maintainers, @n0ts please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
`shipit`

**Comment by gregdek** [comment]
Thanks @n0ts. Since you are one of the maintainers (@n0ts) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
please rebase PR on top of the current devel branch.

`validate_certs` param is not handled anymore in this refactoring. I don't know how we depracate params yet. Coming back this.

File: monitoring/datadog_event.py [hunk 2]
```diff
@@ -30,14 +36,17 @@
 - "Allows to post events to DataDog (www.datadoghq.com) service."
 - "Uses http://docs.datadoghq.com/api/#events API."
 version_added: "1.3"
-author: "Artūras `arturaz` Šlajus (@arturaz)"
+author: "Naoya Nakazawa (@n0ts)"
```
Comment (resmo) [inline monitoring/datadog_event.py]: Please just make a list of authors e.g. 

``` yaml
author: 
  - "Artūras `arturaz` Šlajus (@arturaz)"
  - "Naoya Nakazawa (@n0ts)"
```

File: monitoring/datadog_event.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 #
-# Author: Artūras 'arturaz' Šlajus <x11@arturaz.net>
+# Author: Naoya Nakazawa <naoya.n@gmail.com>
```
Comment (resmo) [inline monitoring/datadog_event.py:5]: please do not remove the previous author.

File: monitoring/datadog_event.py [hunk 4]
```diff
@@ -30,14 +36,17 @@
...
 notes: []
 requirements: []
 options:
     api_key:
         description: ["Your DataDog API key."]
         required: true
         default: null
+    app_key:
+        description: ["Your DataDog app key."]
+        required: true
```
Comment (resmo) [inline monitoring/datadog_event.py]: add a 
~~~yaml
version_added: "2.2"
~~~
below

**Comment by resmo** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @n0ts for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
'ready_for_review'

**Comment by n0ts** [comment]
ready_for_review

**Comment by resmo** [comment]
Thanks @n0ts , to get a green build, this PR needs to be rebased on top of the current devel branch:

```
git checkout devel
git pull git@github.com:ansible/ansible-modules-extras.git devel
git checkout datadog-event-status
git rebase devel
git push -f origin datadog-event-status
```

**Comment by n0ts** [comment]
@resmo Thanks I rebased devel branch.

ready_for_review

**Comment by gregdek** [comment]
Thanks @n0ts. To the current maintainers, @n0ts please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by n0ts** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @n0ts. Since you are one of the maintainers (@n0ts) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: monitoring/datadog_event.py [hunk 5]
```diff
@@ -106,51 +123,41 @@ def main():
...
+                               alert_type=module.params['alert_type'],
+                               aggregation_key=module.params['aggregation_key'],
+                               source_type_name='ansible')
+        if msg['status'] != 'ok':
+            module.fail_json(msg=msg)
 
-    json_body = module.jsonify(body)
-    headers = {"Content-Type": "application/json"}
+        module.exit_json(changed=True, msg=msg)
+    except Exception, e:
```
Comment (resmo) [inline monitoring/datadog_event.py]: Please replace all

``` python
except Exception, e:
```

with

``` python
except Exception:
     e = get_exception()
```

to get a green build.

**Comment by n0ts** [comment]
shipt

**Comment by n0ts** [comment]
shipit

**Pull request merged**